optaplanner-core/src/test/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptorTest.java

```diff
@@ -237,6 +259,103 @@ public void autoDiscoverGetters() {
         assertAllCodesOfCollection(solutionDescriptor.getAllFacts(solution), "otherE1", "p1", "e1", "e2", "v1", "v2");
     }
 
+    @Test
+    public void autoDiscoverFieldsFactCollectionOverridenToSingleProperty() {
+        SolutionDescriptor<TestdataAutoDiscoverFieldOverrideSolution> solutionDescriptor
+                = TestdataAutoDiscoverFieldOverrideSolution.buildSolutionDescriptor();
+        assertMapContainsKeysExactly(solutionDescriptor.getProblemFactMemberAccessorMap(),
+                "singleProblemFact", "listProblemFact");
+        assertMapContainsKeysExactly(solutionDescriptor.getProblemFactCollectionMemberAccessorMap(),
+                "problemFactList");
+        assertMapContainsKeysExactly(solutionDescriptor.getEntityMemberAccessorMap(), "otherEntity");
+        assertMapContainsKeysExactly(solutionDescriptor.getEntityCollectionMemberAccessorMap(),
+                "entityList");
+
+        TestdataObject singleProblemFact = new TestdataObject("p1");
+        List<TestdataValue> valueList = Arrays.asList(new TestdataValue("v1"), new TestdataValue("v2"));
+        List<TestdataEntity> entityList = Arrays.asList(new TestdataEntity("e1"), new TestdataEntity("e2"));
+        TestdataEntity otherEntity = new TestdataEntity("otherE1");
+        List<String> listFact = new TestdataCodeAssertableArrayList<>("list1", Arrays.asList("x", "y"));
+        TestdataAutoDiscoverFieldOverrideSolution solution = new TestdataAutoDiscoverFieldOverrideSolution(
+                "s1", singleProblemFact, valueList, entityList, otherEntity, listFact);
+
+        assertAllCodesOfCollection(solutionDescriptor.getAllFacts(solution),
+                "otherE1", "list1", "p1", "e1", "e2", "v1", "v2");
+    }
+
+    @Test
+    public void autoDiscoverGettersFactCollectionOverridenToSingleProperty() {
```

Comment:
This test duplicates code of previous test. It can be made as one test method that gets solution instance as a parameter.